test/test_vmap.py

```diff
@@ -1150,5 +1160,100 @@ def test_no_random_op_support(self):
                                         'vmap: We do not yet support calling random operations'):
                 vmap(op)(*args)
 
+def construct_v(output, batch_size):
+    return torch.randn(batch_size, *output.shape,
+                       dtype=output.dtype, device=output.device)
+
+def as_tuple(x):
+    if istuple(x):
+        return x
+    elif isinstance(x, list):
+        return tuple(x)
+    else:
+        return x,
+
+def differentiable(args):
+    return tuple(arg for arg in as_tuple(args)
+                 if isinstance(arg, torch.Tensor) and arg.requires_grad)
```

Comment:
I never know which one is supposed to be better `isinstance(arg, torch.Tensor)` or `torch.is_tensor(arg)` ?

Reply:
https://github.com/pytorch/pytorch/blob/0ab83f7f9fcbbc9b7a709b99b38577d14c22a7b7/torch/__init__.py#L226-L231 We seem to recommend that users use `isinstance(arg, torch.Tensor)`.